### PR TITLE
s/point/generic feature

### PR DESCRIPTION
**Pull request by xprogramari**

### CODE REVIEW

File: src/diagonal.works/b6/world.go [hunk 1]
```diff
@@ -15,53 +15,115 @@ import (
...
-	ToGeoJSON() geojson.GeoJSON
+type TagValueType int
+
+const (
+	TagValueTypeString TagValueType = iota
+	TagValueTypeLatLng
+	TagValueTypeBits = 1
+)
+
+type TagValue interface {
```
[the remark this answers is not in the document]
Comment (andreweland) [reply src/diagonal.works/b6/world.go:26]: This is the current canonical list of all types we’ll need to support: 
https://github.com/diagonalworks/diagonal-b6/blob/6e08452688f6c01290ebd2874b3b1553f9e08930/src/diagonal.works/b6/expression.go#L44

File: src/diagonal.works/b6/world.go [hunk 2]
```diff
@@ -242,6 +289,21 @@ func (t *Tags) MergeFrom(other Tags) {
 	}
 }
 
+const LocationTag = "latlng"
```
[the remark this answers is not in the document]
Comment (xprogramari) [reply src/diagonal.works/b6/world.go]: /offline

File: src/diagonal.works/b6/cmd/b6-ingest-gb-uprn/b6-ingest-gb-uprn.go [hunk 3]
```diff
@@ -48,8 +48,8 @@ func main() {
 		}
 	} else {
 		filter = func(c *api.Context, f b6.Feature) (bool, error) {
-			if p, ok := f.(b6.PointFeature); ok {
-				return crop.ContainsPoint(p.Point()), nil
+			if p, ok := f.(b6.Geometry); ok && p.GeometryType() == b6.GeometryTypePoint {
```
[the remark this answers is not in the document]
Comment (xprogramari) [reply src/diagonal.works/b6/cmd/b6-ingest-gb-uprn/b6-ingest-gb-uprn.go:51]: i still dont have a clear idea for collections, so let s cross that bridge when we get there.. b/c i think collections are the only thing that applies to? and yeah, all the features it has will be sort of flattened on the tags side, so maybe we just have a special case for that...

File: src/diagonal.works/b6/ingest/compact/encoding.go [hunk 4]
```diff
@@ -399,22 +399,56 @@ func (lls *LatLngs) UnmarshalWithoutLength(l int, buffer []byte) int {
...
-	value, n := binary.Uvarint(buffer[i:])
-	t.Key = int(key)
-	t.Value = int(value)
-	return i + n
+	t.Key = int(key >> ValueTypeBits)
+	t.ValueType = b6.ValueType(key & ((1 << ValueTypeBits) - 1))
+
+	switch t.ValueType {
+	case b6.ValueTypeString:
+		m := Int(0)
```
Comment (xprogramari) [inline src/diagonal.works/b6/ingest/compact/encoding.go:444]: i hate life :(

i had to pointer b/c of unmarshal ..

**Pull request merged**